### DISCUSSION TITLE
Support Python3.12:  AttributeError: module 'pkgutil' has no attribute 'ImpImporter'.

### DISCUSSION THREAD

**Discussion by ddxv** [apache/superset#25963] [Q&A / Help] [answered]
When trying to install Superset on python3.12 I get the error:
`AttributeError: module 'pkgutil' has no attribute 'ImpImporter'.`

I believe this can be traced back to Numpy version < 1.26.0:
https://github.com/numpy/numpy/issues/23808

Currently apache-superset is pinned to 1.23.5 while the Numpy v1.26.0 was released this past September.  Is it possible to bump numpy?

**Comment by michael-s-molina** [comment, marked as answer]
Hi @Teesmo. Python 12 is not supported yet as you can see [here](https://github.com/apache/superset/blob/ad7bd09819d3fd6ac8f669c6f884551cbd66bcd6/setup.py#L216). Eventually, we'll support it and have a compatible Numpy version 😉

**Reply by stfast** [reply]
After more than half a year and with almost all distributions surpassing python 3.12 version, stucking with numpy 1.23.5 while there is 2.1.0  out, this is more an emberrassment than the answer.

**Reply by ddxv** [reply]
@stfast FYI that link above seems old now, here's a link to the current `pyproject.toml` which also currently shows `Python 3.11` though
